scripts/editor/output-css-variables.js

```diff
@@ -72,6 +72,142 @@ export const globalInner = (itemValues, itemKey) => {
+ *
+ * @return {array}
+ */
+const setBreakpointResponsiveVariables = (attributeVariables, breakpointName, breakpointIndex, numberOfBreakpoints) => {
+	return attributeVariables.map((attributeVariablesObject) => {
+
+		// Calculate default breakpoint index based on order of the breakpoint, inverse property and number of properties in responsiveAttributeObject.
+		const defaultBreakpointIndex = attributeVariablesObject.inverse ? 0 : (numberOfBreakpoints - 1);
+
+		// Expanding an object with an additional breakpoint property.
+		return {
+			...attributeVariablesObject,
+			breakpoint: breakpointIndex === defaultBreakpointIndex ? 'default' : breakpointName,
+		};
+	})
+}
+
+/**
+ * Iterating through variables matching the keys from responsiveAttributes and translating it to responsive attributes names.
+ *
+ * @param {object} responsiveAttributes - Responsive attributes that are read from component's/block's manifest.
+ * @param {object} variables            - Object containing objects with component's/block's attribute variables that are read from manifest.
+ *
+ * @return {object} Object prepared for setting all the variables to its breakpoints.
+ */
+const setupResponsiveVariables = (responsiveAttributes, variables) => {
+	// Iterate through responsive attributes.
+	return Object.entries(responsiveAttributes).reduce((responsiveAttributesVariables, [responsiveAttributeName, responsiveAttributeObject]) => {
+
+		// If responsive attribute doesn't exist in variables object, skip it
+		if (!responsiveAttributeName || _.isEmpty(variables[responsiveAttributeName])) {
+			return responsiveAttributesVariables;
+		}
+
+		// Used for determination of default breakpoint.
+		const numberOfBreakpoints = Object.entries(responsiveAttributeObject).length;
+
+		// Iterate each responsive attribute object as breakpoint name is the key of the object, and value represents the name of the responsive variable.
+		const responsiveAttributeVariables = Object.entries(responsiveAttributeObject).reduce((responsiveAttribute, [breakpointName, breakpointVariableName], breakpointIndex) => {
+			let breakpointVariables = {};
+
+			// Array represents direct value(default or value).
+			if (Array.isArray(variables[responsiveAttributeName])) {
+				breakpointVariables = setBreakpointResponsiveVariables(variables[responsiveAttributeName], breakpointName, breakpointIndex, numberOfBreakpoints);
+
+			// Object treatment goes depending on a value inserted(multiple choice, boolean or similar).
+			} else {
+
+				// Iterate options/multiple choices/boolean...
+				breakpointVariables = Object.entries(variables[responsiveAttributeName])
+				.reduce((acc, [attributeValue, attributeObject]) => {
+					return {
+						...acc,
+						[attributeValue]: setBreakpointResponsiveVariables(attributeObject, breakpointName, breakpointIndex, numberOfBreakpoints),
+					}
+				}, {});
+			}
```

Comment:
This seems like it could be inverted and else could be removed (unnecessary else)

Reply:
I don't see how else could be removed.

If I don't have an object I can't do `Object.entries(...)`. If I don't have an array, `setBreakpointResponsiveVariables` would receive an object instead of an array, and the function would break on that part.

Reply:
I mean, I can do something like this, but I don't see the point in writing two of the exact same returns just to avoid one else.

```
		const responsiveAttributeVariables = Object.entries(responsiveAttributeObject).reduce((responsiveAttribute, [breakpointName, breakpointVariableName], breakpointIndex) => {
			let breakpointVariables = {};

			if (Array.isArray(variables[responsiveAttributeName])) { // Array represents direct value(default or value).
				breakpointVariables = setBreakpointResponsiveVariables(variables[responsiveAttributeName], breakpointName, breakpointIndex, numberOfBreakpoints);
				return {
					...responsiveAttribute,
					[breakpointVariableName]: breakpointVariables,
				};
			}

			// Object treatment goes depending on a value inserted(multiple choice, boolean or similar).
			// Iterate options/multiple choices/boolean...
			breakpointVariables = Object.entries(variables[responsiveAttributeName])
			.reduce((acc, [attributeValue, attributeObject]) => {
				return {
					...acc,
					[attributeValue]: setBreakpointResponsiveVariables(attributeObject, breakpointName, breakpointIndex, numberOfBreakpoints),
				}
			}, {});

			// Collect all the values from one responsive attribute to one object.
			return {
				...responsiveAttribute,
				[breakpointVariableName]: breakpointVariables,
			};
		}, {});
```

But if that's a better practice, or it improves readability, it's ok with me.

Reply:
Sure leave it as is, it's super complicated to read, that's what was worrying me the most. Just good practices than anything.

Reply:
@dingo-d totally agree, but as returning null would change the output of the `.reduce()`, it would be a step back. Also it definitely is a bit hard to read, but I'm sure 3 reduces are still just as easy to read as the mess that would replace them would be.

Reply:
I shortened it and added 2 of the same return sentences 🙂